contracts/src/bridge/L1Oracle.sol

```diff
@@ -98,6 +100,7 @@ contract L1Oracle is Initializable, UUPSUpgradeable, OwnableUpgradeable, Pausabl
         stateRoot = _stateRoot;
         l1FeeOverhead = _l1FeeOverhead;
         l1FeeScalar = _l1FeeScalar;
+        prevStateRoots[uint8(number % 256)] = _stateRoot;
```

Comment:
use an array?

Reply:
I think mapping is more gas efficient.
But also think that we should only have a stateRoots mapping, not stateRoot and prevStateRoots